xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -170,6 +187,59 @@ void watchConfigData(String hostName, int port, ConfigWatcher watcher) {
     adsStream.sendXdsRequest(ADS_TYPE_URL_LDS, ImmutableList.of(ldsResourceName));
   }
 
+  @Override
+  void watchEndpointData(String clusterName, EndpointWatcher watcher) {
```

Comment:
`checkNotNull` for both arguments?

Reply:
Added check for `watcher`. Not needed for `clusterName`. If it is `null`, it crashes immediately in next a few lines (as map key does not allow `null`).